hbase-server/src/main/java/org/apache/hadoop/hbase/master/MobFileCleanerChore.java

```diff
@@ -0,0 +1,289 @@
+          }
+          // Add MOB refs for current region/family
+          allActiveMobFileName.addAll(regionMobs);
+        } // END column families
+      } // END regions
+      // Check if number of MOB files too big (over 1M)
+      if (allActiveMobFileName.size() > 1000000) {
+        LOG.warn("Found too many active MOB files: {}, this may result in high memory pressure.",
+          allActiveMobFileName.size());
+      }
+      LOG.debug("Found: {} active mob refs", allActiveMobFileName.size());
+      allActiveMobFileName.stream().forEach(LOG::debug);
+
+      // Now scan MOB directories and find MOB files with no references to them
+      for (ColumnFamilyDescriptor hcd : list) {
+        List<Path> toArchive = new ArrayList<Path>();
+        String family = hcd.getNameAsString();
+        Path dir = MobUtils.getMobFamilyPath(conf, table, family);
+        RemoteIterator<LocatedFileStatus> rit = fs.listLocatedStatus(dir);
+        while (rit.hasNext()) {
+          LocatedFileStatus lfs = rit.next();
+          Path p = lfs.getPath();
+          if (!allActiveMobFileName.contains(p.getName())) {
+            // MOB is not in a list of active references, but it can be too
+            // fresh, skip it in this case
+            long creationTime = fs.getFileStatus(p).getModificationTime();
+            if ( creationTime < maxCreationTimeToArchive) {
+              /* DEBUG */ LOG.info(
+                " Archiving MOB file{} creation time=" + (fs.getFileStatus(p).getModificationTime()), p);
+              toArchive.add(p);
+            } else {
+              LOG.info("Skipping fresh file: {}", p);
+            }
+          } else {
+            LOG.info("Keepeing active MOB file: {}", p);
+          }
+        }
+        LOG.info(" MOB Cleaner found {} files for family={}", toArchive.size() , family);
+        removeMobFiles(conf, table, family.getBytes(), toArchive);
+        LOG.info(" MOB Cleaner archived {} files", toArchive.size());
+      }
+    }
+  }
+
+
+  /**
+   * Archives the mob files.
+   * @param conf The current configuration.
+   * @param tableName The table name.
+   * @param family The name of the column family.
+   * @param storeFiles The files to be archived.
+   * @throws IOException
+   */
+  public void removeMobFiles(Configuration conf, TableName tableName, byte[] family,
+      List<Path> storeFiles) throws IOException {
+
+    if (storeFiles.size() == 0) {
+      // nothing to remove
+      LOG.debug("Skipping archiving old MOB file: collection is empty");
+      return;
+    }
+    Path mobTableDir = FSUtils.getTableDir(MobUtils.getMobHome(conf), tableName);
+    FileSystem fs = storeFiles.get(0).getFileSystem(conf);
+    Path storeArchiveDir = HFileArchiveUtil.getStoreArchivePath(conf, MobUtils.getMobRegionInfo(tableName),
+      mobTableDir, family);
+
+    for (Path p : storeFiles) {
+      Path archiveFilePath = new Path(storeArchiveDir, p.getName());
+      if (fs.exists(archiveFilePath)) {
```

Comment:
we shouldn't need to do this check, the HFileArchiver already takes care of it.

Reply:
Fixed.